### ISSUE TITLE
Proposal: Support for error types

### ISSUE THREAD

**Issue by jurajkulich** [caarlos0/env#238]
Hello, 
At the moment, it's not possible to differentiate errors via error types. This could help determine the error directly instead of using, e.g. `strings.Contains`.

For example:
```
if err := env.Parse(c); err != nil {
	if err == env.ErrMissingRequired {
	}
	if err == env.ErrEmpty {
	}
}
```

**Comment by caarlos0** [comment]
ah, this is a good one indeed

yes @akutuev feel free to take a hit at it 

thanks everyone!

**Comment by akutuev** [comment]
Hello all,

I did some brief investigations and let me put my input here: Due to our new improvement #208 (and its PR #233) we aggregate multiple errors (even with multiple types) to be able to provide a single aggregated message for a user. Therefore to identify the error type we have to iterate over the list of errors.

If it makes sense we should:
1. Covert type of the Parse/ParseWithFunc error as AggregateError to get access to the list of errors
2. Iterate over errors to get a specific type of the error

In general, this approach won't lead to breaking changes but might bring some complexity

@caarlos0 @jurajkulich Happy to get your opinion

**Comment by caarlos0** [comment]
@akutuev that sounds good yes! 

I wonder if we should wait to see how the aggregate errors are looking like in go 1.20... haven't looked much into it yet, but maybe it makes this easier...

**Comment by caarlos0** [comment]
merged #240
